includes/AMP/Traits/Sanitization_Utils.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Trait Sanitization_Utils.
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\AMP\Traits;
+
+use DOMElement;
+use Google\Web_Stories_Dependencies\AmpProject\Dom\Document;
+use \AmpProject\Dom\Document as AMP_Document;
+
+/**
+ * Trait Sanitization_Utils
+ *
+ * @since 1.1.0
+ */
+trait Sanitization_Utils {
+	/**
+	 * Replaces the HTML start tag to make the language attributes dynamic.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param Document|AMP_Document $document Document instance.
+	 * @return void
+	 */
+	private function transform_html_start_tag( &$document ) {
+		$document->html->setAttribute( 'amp', '' );
+
+		// See get_language_attributes().
+		if ( is_rtl() ) {
+			$document->html->setAttribute( 'dir', 'rtl' );
+		}
+
+		$lang = get_bloginfo( 'language' );
+		if ( $lang ) {
+			$document->html->setAttribute( 'lang', esc_attr( $lang ) );
+		}
+	}
+
+	/**
+	 * Transform all a tags to add target and rel attributes.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param Document|AMP_Document $document Document instance.
+	 * @return void
+	 */
+	private function transform_a_tags( &$document ) {
+		$hyperlinks = $document->getElementsByTagName( 'a' );
+		/* @var DOMElement $hyperlink The <a> element */
```

Comment:
```suggestion
		/** @var DOMElement $hyperlink The <a> element */
```

Reply:
@westonruter Care to elaborate? IDEs seem to pick up the current version already just fine

Reply:
Humm. It seems IDEs are recognizing such comments. In the past my experience is that only comments with `/**` are recognized as phpdoc. 

> ![image](https://user-images.githubusercontent.com/134745/95917846-b84c6280-0d5f-11eb-8da7-a8a050bc040b.png)

Per https://docs.phpdoc.org/latest/guides/docblocks.html

> ![image](https://user-images.githubusercontent.com/134745/95917917-d5813100-0d5f-11eb-8b8c-138672ca5278.png)

See https://docs.phpdoc.org/latest/glossary.html#term-structural-element

Reply:
That means adding a short comment as well then

Reply:
Does it? See the last example:

> ![image](https://user-images.githubusercontent.com/134745/95921449-2ac04100-0d66-11eb-92b4-382288ba9a20.png)

Reply:
PHOCS complains otherwise

Reply:
We [exclude](https://github.com/ampproject/amp-wp/blob/b8f2a48a98da55c77ecad7968954777d9ee39949/.phpcs.xml.dist#L12) that PHPCS complaint:

```xml
<exclude name="Generic.Commenting.DocComment.MissingShort" />
```